plasmapy/diagnostics/charged_particle_radiography.py

```diff
@@ -1115,118 +1266,375 @@ def max_deflection(self):
 
-            # The factor of 5 here is somewhat arbitrary: we just want a
-            # region a few times bigger than the image of the grid on the
-            # detector, since particles could be deflected out
-            size = 5 * np.array([[-w, w], [-w, w]]) * u.m
+            # dE is in MeV
+            dE = interpolated_stopping_power * layer.thickness.to(u.cm).value
 
-        # Generate the histogram
-        intensity, h, v = np.histogram2d(
-            xloc, yloc, range=size.to(u.m).value, bins=bins
-        )
+            # If dE > E for a given energy, set dE=E (stop the particle)
+            dE = np.where(dE > energies, energies, dE)
 
-        # h, v are the bin edges: compute the centers to produce arrays
-        # of the right length (then trim off the extra point)
-        h = ((h + np.roll(h, -1)) / 2)[0:-1]
-        v = ((v + np.roll(v, -1)) / 2)[0:-1]
+            energies += -dE
 
-        # Throw a warning if < 50% of the particles are included on the
-        # histogram
-        percentage = np.sum(intensity) / self.nparticles
-        if percentage < 0.5:
-            warnings.warn(
-                f"Only {percentage:.2%} of the particles are shown "
-                "on this synthetic radiograph. Consider increasing "
-                "the size to include more.",
-                RuntimeWarning,
+            deposited[i, :] = dE
+
+        # Normalize the deposited energy array so that each number represents
+        # the fraction of a population of particles of that energy stopped
+        # in that layer.
+        deposited = deposited / np.sum(deposited, axis=0)
+
+        # If this flag is set, return only the layers that correspond to active
+        # medium, ignoring the filter and substrate layers
+        if return_only_active:
+            active_ind = [i for i in range(len(self._layers)) if self._layers[i].active]
+            deposited = deposited[active_ind, :]
+
+        return deposited
+
+    def energy_bands(self, energy_range, dE, return_only_active=True):
+        """
+        Calculate the energy bands in each of the active layers of a film stack.
+
+        Parameters
+        ----------
+        stack : list of Layer objects
+            This list of Layer objects defines the composition of the film stack.
+
+        energy_range : `~astropy.units.Quantity` list, shape [2,]
+            A range of energies to include in the calculation. Units convertable
+            to eV.
+
+        dE :  `~astropy.units.Quantity`
+            Spacing between energy bins in the calculation. Units convertable
+            to eV.
+
+        return_only_active : boolean, optional
+            If True, only the energy bands of layers in which the `active` property
+            is True will be returned. This is usually desirable, since particles
+            captured in other layers will not be measured. If False, energy bands in
+            all layers of the stack are returned. The default is True.
```

Comment:
```suggestion
        return_only_active : `bool`, optional
            If `True`, only the energy bands of layers in which the `active` property
            is `True` will be returned. This is usually desirable, since particles
            captured in other layers will not be measured. If `False`, energy bands in
            all layers of the stack are returned. The default is `True`.
```